dependency/retrieval/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"log"
+	"os"
+
+	"github.com/joshuatcasey/libdependency/retrieve"
+	"github.com/joshuatcasey/libdependency/versionology"
+	"github.com/paketo-buildpacks/go-dist/dependency/retrieval/alternative"
+	"github.com/paketo-buildpacks/packit/v2/cargo"
+)
+
+// libdependency implementation
+func main() {
+	fetcher := NewReleaseFetcher()
+
+	versionFetcherFunc := func() (versionology.VersionFetcherArray, error) {
+		releases, err := fetcher.List()
+		if err != nil {
+			return nil, err
+		}
+
+		var versions []string
+		for _, release := range releases {
+			versions = append(versions, release.SemVer.String())
+		}
+
+		return versionology.NewSimpleVersionFetcherArray(versions...)
+	}
+
+	generateMetadataFunc := func(version versionology.VersionFetcher) (versionology.Dependency, error) {
+		release, err := fetcher.Get(version.Version().String())
```

Comment:
To me, the types here would be the hardest thing to digest when using `libdependency`.
Specifically around `version.Version().String()`, having to run the `Version()` function on a variable already named `version`, and then converting to a string is confusing. (I get the code in libdependency, I'm just noting that from looking at the implementation it's not the most intuitive)